cns/restserver/metrics.go

```diff
@@ -10,6 +10,12 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
+const (
+	AssignIPConfigFailure   = "assign_ip_config_failure"
+	PopulateIPConfigFailure = "populate_ip_config_failure"
+	NoIPsAvailableFailure   = "no_ips_available_failure"
```

Comment:
last nit, and then I think this is good to go - these vars are not used outside of this package and so should not be exported
```suggestion
	assignIPConfigFailure   = "assign_ip_config_failure"
	populateIPConfigFailure = "populate_ip_config_failure"
	noIPsAvailableFailure   = "no_ips_available_failure"
```